packages/nala/build.sh

```diff
@@ -2,72 +2,31 @@ TERMUX_PKG_HOMEPAGE=https://gitlab.com/volian/nala
 TERMUX_PKG_DESCRIPTION="Commandline frontend for the apt package manager"
 TERMUX_PKG_LICENSE="GPL-3.0"
 TERMUX_PKG_MAINTAINER="@termux"
-TERMUX_PKG_VERSION=0.7.2
+TERMUX_PKG_VERSION=0.11.1
 TERMUX_PKG_SRCURL=https://gitlab.com/volian/nala/-/archive/v${TERMUX_PKG_VERSION}/nala-v${TERMUX_PKG_VERSION}.tar.bz2
-TERMUX_PKG_SHA256=0ac394ab2bd924c28c513a8e8bfc9e899a21f2f8fa5e5d77366cb20e0be8676e
+TERMUX_PKG_SHA256=4ce87d75e785e45b1638817077797a3fcd4a15059615455fc1af0127c70cf5cd
 TERMUX_PKG_DEPENDS="python-apt"
 TERMUX_PKG_PLATFORM_INDEPENDENT=true
 TERMUX_PKG_BUILD_IN_SRC=true
 
 _PYTHON_VERSION=$(. $TERMUX_SCRIPTDIR/packages/python/build.sh; echo $_MAJOR_VERSION)
 
-TERMUX_PKG_RM_AFTER_INSTALL="
-lib/python${_PYTHON_VERSION}/site-packages/__pycache__
-lib/python${_PYTHON_VERSION}/site-packages/easy-install.pth
-lib/python${_PYTHON_VERSION}/site-packages/site.py
-"
-
-termux_step_post_get_source() {
-	sed 's:@TERMUX_PKG_VERSION@:'${TERMUX_PKG_VERSION##*:}':g' \
-		${TERMUX_PKG_BUILDER_DIR}/pyproject.toml.diff | \
-		patch --silent -p1
-}
-
-termux_step_pre_configure() {
-	termux_setup_python_crossenv
-	pushd $TERMUX_PYTHON_CROSSENV_SRCDIR
-	_CROSSENV_PREFIX=$TERMUX_PKG_BUILDDIR/python-crossenv-prefix
-	python${_PYTHON_VERSION} -m crossenv \
-		$TERMUX_PREFIX/bin/python${_PYTHON_VERSION} \
-		${_CROSSENV_PREFIX}
-	popd
-	. ${_CROSSENV_PREFIX}/bin/activate
-
-	build-pip install tomli
-}
-
 termux_step_make_install() {
-	python setup.py install --force
-
-	export PYTHONPATH=$TERMUX_PREFIX/lib/python${_PYTHON_VERSION}/site-packages
-	python setup.py install --force --prefix $TERMUX_PREFIX
-
-	pushd $PYTHONPATH
-	_NALA_EGG=
-	for f in nala-${TERMUX_PKG_VERSION##*:}-py${_PYTHON_VERSION}.egg; do
-		if [ -e "$f" ]; then
-			_NALA_EGG="$f"
-			break
-		fi
-	done
-	test -n "${_NALA_EGG}"
-	popd
+	# Activate venv to trick python into installing in the correct location
+	python -m venv $TERMUX_PKG_BUILDDIR/python-venv
+	source $TERMUX_PKG_BUILDDIR/python-venv/bin/activate
 
+	python${_PYTHON_VERSION} -m pip install . --no-deps --prefix $TERMUX_PREFIX
 	install -Dm600 -t $TERMUX_PREFIX/etc/nala debian/nala.conf
 }
 
 termux_step_create_debscripts() {
 	cat <<- EOF > ./postinst
 	#!$TERMUX_PREFIX/bin/sh
-	echo "./${_NALA_EGG}" >> $TERMUX_PREFIX/lib/python${_PYTHON_VERSION}/site-packages/easy-install.pth
 	mkdir -p $TERMUX_PREFIX/var/lib/nala
 	mkdir -p $TERMUX_PREFIX/var/log/nala
+	mkdir -p $TERMUX_PREFIX/var/lock
 	echo "Installing dependencies through pip..."
-	pip3 install anyio httpx jsbeautifier pexpect rich
-	EOF
-
-	cat <<- EOF > ./prerm
-	#!$TERMUX_PREFIX/bin/sh
-	sed -i "/\.\/${_NALA_EGG//./\\.}/d" $TERMUX_PREFIX/lib/python${_PYTHON_VERSION}/site-packages/easy-install.pth
+	pip3 install anyio httpx jsbeautifier pexpect rich tomli python-debian
```

Comment:
Seems `typing` (and potentially other dependencies) are missing here

Reply:
There are only 2 "missing" dependencies:
- `typer` is automatically installed because its a dependency of `rich`.
- `typing-extentions`, which is a backport of `typing` features, only needed for python <3.10 when using nala, and wasn't added here because that's the version on the termux repos.

I'll add them anyway, since we may not know if the user is using the latest version of python